utoipa-gen/src/schema.rs

```diff
@@ -56,38 +63,15 @@ impl<'a> ComponentPart<'a> {
         }
     }
 
-    fn from_ident(ty: &'a Ident) -> ComponentPart<'a> {
-        ComponentPart {
-            child: None,
-            generic_type: None,
-            ident: ty,
-            value_type: if ComponentType(ty).is_primitive() {
-                ValueType::Primitive
-            } else {
-                ValueType::Object
-            },
-        }
-    }
-
-    fn from_type_path(
-        type_path: &'a TypePath,
-        op: impl Fn(&'a Ident, &'a PathSegment) -> ComponentPart<'a>,
-        or_else: impl Fn(&'a PathSegment) -> ComponentPart<'a>,
-    ) -> ComponentPart<'a> {
-        let segment = type_path
-            .path
-            .segments
-            .pairs()
-            .find_map(|pair| match pair {
-                Pair::Punctuated(_, _) => None,
-                Pair::End(segment) => Some(segment),
-            })
-            .unwrap();
-
-        if segment.arguments.is_empty() {
-            op(&segment.ident, segment)
+    /// Creates a [`ComponentPath`] from a [`TypePath`].
+    /// `op` is a function
```

Comment:
I guess now this is unnecessary line :)
```suggestion
    /// `op` is a function
```